src/lib/support/BufferWriter.cpp

```diff
@@ -23,10 +23,21 @@ namespace Encoding {
 BufferWriter & BufferWriter::Put(const char * s)
 {
     static_assert(CHAR_BIT == 8, "We're assuming char and uint8_t are the same size");
-    while (*s != 0)
-    {
-        Put(static_cast<uint8_t>(*s++));
-    }
+    return Put(s, strlen(s));
+}
+
+BufferWriter & BufferWriter::VFormat(const char * format, va_list args)
+{
+    mNeeded += vsnprintf(reinterpret_cast<char *>(mBuf + mNeeded), Available(), format, args);
```

Comment:
```
The number of characters that would have been written if n had been sufficiently large, not counting the terminating null character.
If an encoding error occurs, a negative number is returned.
Notice that only when this returned value is non-negative and less than n, the string has been completely written.
```

Reply:
Good catch on the negative number ... did not read that. Updated.

Also moved that only stringbuilder has printf, not bufferwriter.